### PR TITLE
feat: update fundamental-styles to v0.12.0-rc.18

### LINKED ISSUE #646: Tree: expand collapse not working with functional components
### Versions
**fundamental-react:** 0.6.1

The props `onExpandClick`, `expandData` and `isExpanded` are not being passed properly down through `DetailTree`.  I believe this has to do with our `React.cloneElement` functions, we might need to add something recursive in there. 

https://github.com/SAP/fundamental-react/blob/master/src/TreeView/TreeView.js#L120

`child` here is empty if using a pass through component.

Not working: 
```
const DetailTree = () => (
     <TreeView.Item>						
          <TreeView.Row>
                    <TreeView.Col>Row added through a custom Component: expand collapse- not w</TreeView.Col>
          </TreeView.Row>	

          <TreeView.Branch>							
               <TreeView.Item>							  
                    <TreeView.Row>
                         <TreeView.Col>Generated Child</TreeView.Col>
                    </TreeView.Row>                                
               </TreeView.Item>							  
          </TreeView.Branch>	

     </TreeView.Item>                    
)

class TreeNotWorking extends React.Component {
     
     render() {
          return (
               <TreeView>                
                    <TreeView.Tree>                       
                    <DetailTree />  
                    </TreeView.Tree>
               </TreeView>
          );
     }     
}

export default TreeNotWorking;
```

Working: 
```
class TreeWorking extends React.Component {
     
render() {
    return (
        <TreeView>                
            <TreeView.Tree> 
                <TreeView.Item>	                         
                        <TreeView.Row>
                            <TreeView.Col>Row added directly in render</TreeView.Col>
                        </TreeView.Row>	

                        <TreeView.Branch>							
                            <TreeView.Item>							  
                                <TreeView.Row>
                                    <TreeView.Col>Hardcoded row child</TreeView.Col>
                                </TreeView.Row>                                
                            </TreeView.Item>							  
                        </TreeView.Branch>
                </TreeView.Item> 
            </TreeView.Tree>
        </TreeView>  
        );
    }
}

export default TreeWorking;
```

_**DISCLAIMER:**
After triaging an issue, the fundamental-react team will see if it can be reproduced or confirmed. If more information is needed, the fundamental-react team will contact the author.  Any issues awaiting responses from the author for more than 7 days will be closed.  The author can re-open the issue at a later time if they can present the requested information._

### PR DESCRIPTION
**Pull request by prsdthkr**
### Description
BREAKING CHANGE:
In this change we
* Update `fundamental-styles` to `0.12.0-rc.18` and support the new features and fixes therein.
  * Removed `LayoutGrid` component in favor of the new grid system based on `Container`, `Row`, and `Column` components
  * Remove `LayoutPanel.colSpan` property. Added story for `LayoutPanel` in the new grid system.
  * Deprecated the `TreeView`component in favor of the new `Tree` component.
  * Shellbar avatar now has a branded border.
* Storybook
  * update storybook to v6.0.16
  * fix missing storyDescriptions and component descriptions.
  * Add copy button to easily copy import statements to clipboard.

### Question
* `TreeView` styles are broken and I've marked it as deprecated. Do we want to remove it entirely instead? 
  Yes

Fixes #646  #310

### CODE REVIEW

**Comment by jbadan** [comment]
Re: treeview - let's just delete it.

File: .storybook/custom/components/CopyButton.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import Button from '../../../src/Button/Button';
+
+const CopyButton = ({ copyText }) => {
```
Comment (jbadan) [inline .storybook/custom/components/CopyButton.js:4]: 👏

File: .storybook/custom/components/DocsPage.js [hunk 2]
```diff
@@ -40,14 +40,42 @@ const DocsPage = () => {
...
     return (
         <>
         <Header />
         <Title />
         <Toc />
         <Subtitle />
+        {context?.parameters?.deprecated && <Description desc={context?.parameters?.deprecated} />}
+        {!context?.parameters?.noImport && showImport()}
+        {context?.parameters?.displaySubComponentImports && showSubImports()}
         {context?.parameters?.description && <Description desc={context?.parameters?.description} />}
```
Comment (jbadan) [inline .storybook/custom/components/DocsPage.js:78]: Not caused by this pr (it's in master too) - but our descriptions are gone 😢

Comment (prsdthkr) [reply .storybook/custom/components/DocsPage.js:78]: I can see the description for our Tree component here: https://deploy-preview-1172--fundamental-react.netlify.app/?path=/docs/component-api-tree--wrapping-and-non-wrapping-contents

Comment (jbadan) [reply .storybook/custom/components/DocsPage.js]: I got the story descriptions working here, but not the component descriptions: https://github.com/SAP/fundamental-react/pull/1189

**Comment by prsdthkr** [comment]
Closing and reopening to trigger checks.

**Pull request merged**